From 34b6e24316e83bb13bb1b420616bc986783bb249 Mon Sep 17 00:00:00 2001
From: Jonathan Thomas <echidnaman@kubuntu.org>
Date: Sat, 22 Oct 2011 19:31:55 -0400
Subject: [PATCH 16/20] Autorestart when we crash. This way a polkit-kde-1
 crash won't prevent polkit auth from working until
 the next login.

---
 main.cpp |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/main.cpp b/main.cpp
index e197cee..2fa928f 100644
--- a/main.cpp
+++ b/main.cpp
@@ -21,6 +21,7 @@
 #include <KCmdLineArgs>
 #include <KAboutData>
 #include <KLocale>
+#include <KCrash>
 
 #include "policykitkde.h"
 
@@ -32,6 +33,8 @@ int main(int argc, char *argv[])
     aboutData.addAuthor(ki18n("Jaroslav Reznik"), ki18n("Maintainer"), "jreznik@redhat.com");
     aboutData.setProductName("policykit-kde/polkit-kde-authentication-agent-1");
 
+    KCrash::setFlags(KCrash::AutoRestart);
+
     KCmdLineArgs::init(argc, argv, &aboutData);
 
     if (!PolicyKitKDE::start()) {
-- 
1.7.7.4

